oeps/oep-0045-arch-ops-and-config.rst

```diff
@@ -0,0 +1,169 @@
+
+    Installing and running an Open edX instance is not simple. We strongly recommend that you use a service provider to run the software for you. 
+
+The proposal below outlines how we can create a cleaner, more intuitive interface for operating Open edX and in doing so help both edX.org and the Open edX community at large achieve better outcome at a faster pace.  In brief it is:
+
+* Use Dockerfiles to capture system dependencies and operate the resulting images in all environments.
+* Simplify and standardize how to configure an application.
+* Document standard operations.
+
+ 
+Containers
+**********
+
+**Why Containers**
+
+For managing microservice architectures containers are an industry standard for the following reasons: 
+
+* **Increased portability.**  Run on any host OS that supports containers.
+* **Greater efficiency.**  Ability to safely run multiple containers on the same hardware allows cost savings through efficient use of resources.
+* **Improved security.** Containers improve security by isolating applications from the host system and from each other.
+* **Improved application development.**  Allows developers to run a more production-like environment locally, preventing discrepancies between environments.
+* **Speed.** Start, create, replicate or destroy containers in seconds. Increases velocity for development, deployment, and production operations.
+* **Operational simplicity/consistency.**  By isolating application processes from the host OS it is simpler to manage and maintain the host OS.  Homogeneous administration of heterogeneous components, reduces the range of skill sets required to operate environments.
+
+
+**Clear System Dependency Documentation**
+
+For operators who don’t want to run in Docker containers the Dockerfile acts as documentation for the system dependencies they will need to recreate.  Everything an operator needs to know about how to set up their environment is all in one place rather than distributed across multiple overlapping and overriding files.
+
+In order to function as documentation for operators Dockerfiles will be well-commented and use `native Dockerfile syntax`_ to describe the image.  No Ansible should be run as part of the image build. Similarly, bash scripts should be avoided. If they must be used due to a limitation of the Dockerfile commands, the scripts must live within the same codebase as the Dockerfile.
+
```

Comment:
Users must be able to rebuild the production and development-ready images locally. This means that:

1. the Dockerfile should not point to private pypi/npm/git repos
2. the default values of the Docker ARGuments should not have to be modified for most users

(btw ARG values should be documented, in addition to runtime configuration settings)

Reply:
Agreed - I would strongly encourage us to ensure that 'docker build .' with no arguments produces a valid image for most users.

Reply:
Agreed with @regisb

Reply:
Good point, thanks.